### DISCUSSION TITLE
Add the ability to trasnform/tween difficulty settings over time

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#13009] [Ideas]
I think it would be really cool to see maps in osu!std similar to SV maps in osu!mania. That is, maps where the AR (or SV in mania) of small sections of the song can change. 

This isn't the same as individual notes being able to have different AR though, because with that scenario the AR of an individual note doesn't change as the note gets closer, while with SV in mania it does. 

How it might work: 
In my opinion it should be a seperate thing from BPM/SV, unlike in mania. Having it as a seperate thing means you can have changing BPM without having to worry about the AR changing, and normal maps will therefore be unaffected by this change.

I think it should be another row of multipliers just like SV multiplier in osu!std, available in the F6 menu while mapping. I think it could be called AR multiplier.

**Comment by LastExceed** [comment]
iirc peppy already announced that per-hitobject AR/CS/OD/ect are going to be a thing at some point

**Comment by ghost** [comment]
> iirc peppy already announced that per-hitobject AR/CS/OD/ect are going to be a thing at some point

Yes, which is why i mentioned that this is *not* the same as that lol

**Comment by abraker95** [comment]
Wait how is it not the same? Per-hitobject AR or individual hitobject AR is what you are suggesting.

**Comment by ghost** [comment]
It's not the same as that, because with AR changing as the map progresses, the AR of an individual note can change as the ring gets closer to the circle.

An extreme example:
let's say the base AR is 10

the AR multiplier could be -1.0 for a second of AR-10, near-stopping all approach circles from moving for a second.
After that it could be set to 1.29x until the next note, making it AR12.9, meaning the next note is near-instant.
Then it could continue as 1.0x or maybe switch from -1.0x and 1.29x every note.

[Here's](https://www.youtube.com/watch?v=GNmHxtLVLAI) an example of what an SV map can look like in osu!mania

**Comment by abraker95** [comment]
I see. It's more like dynamic timewarp affecting only approach circles rather than AR on individual notes. AR on individual notes would vary the time it takes for the approach circles to close, but they close in at some constant rate. This would make all approach circles speed up or slow down at various times depending on how and when the rate is changed. 

For those who have trouble wrapping their head around this, imagine being able to make approach circles close in a glitch-like motions or have approach circles stop mid way before a sudden close in.

**Comment by MitsukaiDokuro** [comment]
Variable beatmap stats would greatly increase the toolset of mappers, though I would love to see it implemented explicitly rather than with an arbitrary multiplier.
